iceoryx_binding_c/source/c_publisher.cpp

```diff
@@ -31,17 +31,28 @@ extern "C" {
 #include "iceoryx_binding_c/publisher.h"
 }
 
+void iox_pub_options_init(iox_pub_options_t* options)
+{
+    PublisherOptions publisherOptions;
+    options->historyCapacity = publisherOptions.historyCapacity;
```

Comment:
please check if `options` is a nullptr before accessing it

Reply:
I thought about it and can do it but in the end I can get an invalid non-nullptr.

Reply:
I think just printing a warning/error and continue. If it's a nullptr, it will crash later when the user access it on the caller site, but at least there is some output on the console.